exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -0,0 +1,277 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry/sdk'
+require 'net/http'
+
+require 'opentelemetry/proto/common/v1/common_pb'
+require 'opentelemetry/proto/resource/v1/resource_pb'
+require 'opentelemetry/proto/trace/v1/trace_pb'
+require 'opentelemetry/proto/collector/trace/v1/trace_service_pb'
+
+module OpenTelemetry
+  module Exporter
+    module OTLP
+      # An OpenTelemetry trace exporter that sends spans over HTTP as Protobuf encoded OTLP ExportTraceServiceRequests.
+      class Exporter # rubocop:disable Metrics/ClassLength
+        SUCCESS = OpenTelemetry::SDK::Trace::Export::SUCCESS
+        FAILURE = OpenTelemetry::SDK::Trace::Export::FAILURE
+        private_constant(:SUCCESS, :FAILURE)
+
+        # Default timeouts in seconds.
+        KEEP_ALIVE_TIMEOUT = 30
+        OPEN_TIMEOUT = 5
+        READ_TIMEOUT = 5
+        WRITE_TIMEOUT = 5
+        RETRY_COUNT = 5
+        private_constant(:KEEP_ALIVE_TIMEOUT, :OPEN_TIMEOUT, :READ_TIMEOUT, :WRITE_TIMEOUT, :RETRY_COUNT)
+
+        def initialize(endpoint: config_opt('OTEL_EXPORTER_OTLP_SPAN_ENDPOINT', 'OTEL_EXPORTER_OTLP_ENDPOINT', default: 'localhost:55681/v1/trace'), # rubocop:disable Metrics/AbcSize
+                       insecure: config_opt('OTEL_EXPORTER_OTLP_SPAN_INSECURE', 'OTEL_EXPORTER_OTLP_INSECURE', default: false),
+                       certificate_file: config_opt('OTEL_EXPORTER_OTLP_SPAN_CERTIFICATE', 'OTEL_EXPORTER_OTLP_CERTIFICATE'),
+                       headers: config_opt('OTEL_EXPORTER_OTLP_SPAN_HEADERS', 'OTEL_EXPORTER_OTLP_HEADERS'), # TODO: what format is expected here?
```

Comment:
@mwear this allows passing in headers to be added to each request. The format of the environment variable doesn't appear to be specified. What should we support here? I assume we also need to support explicitly passing a `Hash`?

Reply:
I looked into this a bit. JS appears to accept only [an Object](https://github.com/open-telemetry/opentelemetry-js/blob/ff3ee0f6d1a3625e2dbc191501358181e99667fd/packages/opentelemetry-exporter-collector/src/platform/node/CollectorExporterNodeBase.ts#L43) (Hash, for all intents and purposes), and [does validation](https://github.com/open-telemetry/opentelemetry-js/blob/ff3ee0f6d1a3625e2dbc191501358181e99667fd/packages/opentelemetry-exporter-collector/src/util.ts#L25-L38), throwing out any malformed key/value pairs and coercing the values into strings

They don't have an env var option which, i would assume will have to be a string of comma separated `key:value` pairs, ie: `header_one:value_one,header_two:value_two`.

Optimistically i'd say we could support array of k/v pairs, hash, and csv separated string of k/v pairs. realistically it's probably hard to document and cumbersome to validate and we should maybe stick to some subset of those. CSV string and hash feels pretty flexible to me imo, but fine with any choices here.

Reply:
I’ll go with CSV string and hash. I’ll try to get that wrapped up on Tuesday.

Reply:
Done